release/storage/azure.py

```diff
@@ -35,15 +35,29 @@ def copy(self, source_path, destination_path):
         resp = None
         try:
             resp = self.blob_service.copy_blob(self.container, destination_path, az_blob_url)
-        except (azure.common.AzureConflictHttpError, azure.common.AzureException):
+        except (azure.common.AzureException, azure.common.AzureMissingResourceHttpError):
             # Cancel the past copy, make a new copy
-            properties = self.blob_service.get_blob_properties(self.container, destination_path)
-            assert properties.id
-            self.blob_service.abort_copy_blob(self.container, destination_path, properties.id)
-
-            # Try the copy again
-            resp = self.blob_service.copy_blob(self.container, destination_path, az_blob_url)
-
+            # copy_blob returns a returns a copy operation properties object, including a copy ID can be used to check
+            # or abort the  copy operation.
+
+            # copy_blob as per Azure is a best effort operation.
+
+            # If the previous copy failed due to HTTP Connection Error, we can abort it using copy.id and retry.
+            # Otherwise, we simply retry again.
+
+            try:
+                assert resp.id
```

Comment:
Why do we not need the id from the call to `get_blob_properties`?

Reply:
> On the retry, the a wrong attribute was being looked up that resulted in an error AttributeError: 'Blob' object has no attribute 'id' - This was cleared up looking at the Azure Documentation and finding that .id is the property of the copy_blob operation (Ref: https://github.com/Azure/azure-storage-python/blob/master/azure-storage-blob/azure/storage/blob/baseblobservice.py#L2947)

Reply:
doh, thanks